backend/src/apiserver/server/visualization_server.go

```diff
@@ -80,8 +80,8 @@ func (s *VisualizationServer) generateVisualizationFromRequest(request *go_clien
 	serviceURL := s.getVisualizationServiceURL(request)
 	if !isVisualizationServiceAlive(serviceURL) {
 		return nil, util.NewInternalServerError(
-			fmt.Errorf("service not available"),
-			"Service not available",
+			fmt.Errorf("visualization server is not responsible"),
```

Comment:
Thanks! This is better than before.

One more info I wanted is that, when calling `isVisualizationServiceAlive`, it just returns whether alive or not. However, the root cause might come from many different reasons. For example, misconfigured istio setting preventing visualization service being reached.

Therefore, I think it's better refactoring `isVisualizationServiceAlive` to return an error instead. Then, the error we return here can be a wrapper on that error like usage in https://github.com/kubeflow/pipelines/blob/b1c1431eb8d6b7cc63c55f5e828faf9859a50e55/backend/src/apiserver/server/experiment_server.go#L28

If we change it like this, each time when `isVisualizationServiceAlive` fails, we can easily debug what went wrong from the browser instead of looking into logs from different servers.

Reply:
Thanks for explanation!

So apart from using `util.Wrap()`, I tried to use `errors.New()` to bring up sufficient error message in `isVisualizationServiceAlive `.